R/get_taxonlijsten_lists.R

```diff
@@ -0,0 +1,112 @@
+#' @title Query to extract Taxonlijsten from D0156_00_Taxonlijsten
+#'
+#' @description This function queries D0156_00_Taxonlijsten and gives an
+#' overview of all the taxon lists and list versions currently available in the
+#' database. Only the latest version is shown unless specified otherwise
+#'
+#' @param connection dbconnection with the database 'D0156_00_Taxonlijsten'
+#' on the inbo-sql07-prd server
+#' @param list name of the taxonlist that you want to retrieve. Wildcards %
+#' are allowed. Case insensitive.
+#' @param version A choice ('latest', 'old', 'all'). If 'latest' (the default)
+#' only the most recent version is returned. If 'old' all but the most recent
+#' version is returned. If 'all' all versions are returned.
+#' @param collect If FALSE (the default), a remote tbl object is returned. This
+#' is like a reference to the result of the query but the full result of the
+#' query is not brought into memory. If TRUE the full result of the query is
+#' collected (fetched) from the database and brought into memory of the working
+#' environment.
+#'
+#' @return A remote tbl object (collect = FALSE) or a tibble dataframe (collect
+#' = TRUE) with variables TaxonlijstType, TaxonlijstCode, Taxonlijst,
+#' Publicatiejaar, Version, ReferentieURL, Criteria, Validering, Vaststelling.
+#'
+#' @importFrom glue glue_sql
+#' @importFrom DBI dbGetQuery
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr collect tbl sql
+#'
+#' @export
+#' @family taxonlijsten
+#' @examples
+#' \dontrun{
+#' library(inbodb)
+#' con <- connect_inbo_dbase("D0156_00_Taxonlijsten")
+#'
+#' # get the most recent version of the 'Rode lijst van de Dagvlinders'
+#' get_taxonlijsten_lists(con, version = 'latest', list =
+#' '%rode%dagvlinders%', collect = FALSE)
+#'
+#' # get all recent red lists
+#' get_taxonlijsten_lists(con, list = '%rode lijst%')
+#'
+#' # get all taxonlist versions in the database
+#' get_taxonlijsten_lists(con, version = 'all', collect = TRUE)
+#'
+#' # use function with default values (only most recent versions)
+#' get_taxonlijsten_lists(con)
+#'
+#' # status of red lists
+#' RL <- get_taxonlijsten_lists(con, list = '%rode lijst%')
+#' select(RL,"Taxonlijst", "PublicatieJaar", "Criteria", "Validering",
+#' "Vaststelling")
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
```

Comment:
```suggestion
#' rm(con, RL)
```
Due to the dontrun examples are not executed as a test, but if this would be the case, it would be better to remove all objects at the end, so this is just a suggestion for best practice.
And also, if checklist checks for lintr code style, it would only accept snake_case, so no capitals as in `RL`.

Reply:
dropped all objects, all is snake_case now